macOS 10.15.2下编译出错

mv -f .deps/pdnsd-ctl.Tpo .deps/pdnsd-ctl.Po
aarch64-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I../..   -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include   -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/Volumes/OpenWrt/openwrt/build_dir/target-aarch64_cortex-a72_musl/pdnsd-1.2.9b-par:pdnsd-1.2.9b-par -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include  -DCLIENT_ONLY -c ../rr_types.c
In file included from ../cache.h:28:0,
                 from ../helpers.h:32,
                 from ../rr_types.c:30:
../ipvers.h:218:0: warning: "SOL_IPV6" redefined
 # define SOL_IPV6 IPPROTO_IPV6
 
In file included from /Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify/sys/socket.h:22:0,
                 from ../ipvers.h:29,
                 from ../cache.h:28,
                 from ../helpers.h:32,
                 from ../rr_types.c:30:
/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/sys/socket.h:247:0: note: this is the location of the previous definition
 #define SOL_IPV6        41
 
aarch64-openwrt-linux-musl-gcc  -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/Volumes/OpenWrt/openwrt/build_dir/target-aarch64_cortex-a72_musl/pdnsd-1.2.9b-par:pdnsd-1.2.9b-par -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include   -L/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/lib -L/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/lib -L/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/usr/lib -L/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/lib -znow -zrelro  -o pdnsd-ctl pdnsd-ctl.o rr_types.o 
Making all in rc
Making all in RedHat
make[8]: Nothing to be done for `all'.
Making all in SuSE
make[8]: Nothing to be done for `all'.
Making all in Debian
make[8]: Nothing to be done for `all'.
Making all in Slackware
make[8]: Nothing to be done for `all'.
Making all in ArchLinux
make[8]: Nothing to be done for `all'.
make[8]: Nothing to be done for `all-am'.
Making all in test
aarch64-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I../..   -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include   -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/Volumes/OpenWrt/openwrt/build_dir/target-aarch64_cortex-a72_musl/pdnsd-1.2.9b-par:pdnsd-1.2.9b-par -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include  -MT if_up.o -MD -MP -MF .deps/if_up.Tpo -c -o if_up.o if_up.c
In file included from ../cache.h:28:0,
                 from ../helpers.h:32,
                 from if_up.c:5:
../ipvers.h:218:0: warning: "SOL_IPV6" redefined
 # define SOL_IPV6 IPPROTO_IPV6
 
In file included from /Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify/sys/socket.h:22:0,
                 from ../ipvers.h:29,
                 from ../cache.h:28,
                 from ../helpers.h:32,
                 from if_up.c:5:
/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/sys/socket.h:247:0: note: this is the location of the previous definition
 #define SOL_IPV6        41
 
mv -f .deps/if_up.Tpo .deps/if_up.Po
aarch64-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I../..   -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/usr/include -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify -I/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include   -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/Volumes/OpenWrt/openwrt/build_dir/target-aarch64_cortex-a72_musl/pdnsd-1.2.9b-par:pdnsd-1.2.9b-par -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/usr/include   -c ../netdev.c
In file included from ../netdev.c:59:0:
../ipvers.h:218:0: warning: "SOL_IPV6" redefined
 # define SOL_IPV6 IPPROTO_IPV6
 
In file included from /Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/fortify/sys/socket.h:22:0,
                 from ../ipvers.h:29,
                 from ../netdev.c:59:
/Volumes/OpenWrt/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-7.5.0_musl/include/sys/socket.h:247:0: note: this is the location of the previous definition
 #define SOL_IPV6        41
 
../netdev.c: In function 'is_local_addr':
../netdev.c:78:17: error: 'struct sockaddr' has no member named 'sa_len'
  ((ifr).ifr_addr.sa_len > sizeof(struct sockaddr) ? \
                 ^
../netdev.c:339:10: note: in expansion of macro 'SIZEOF_ADDR_IFREQ'
   cnt += SIZEOF_ADDR_IFREQ(*ir);
          ^~~~~~~~~~~~~~~~~
../netdev.c:80:18: error: 'struct sockaddr' has no member named 'sa_len'
    (ifr).ifr_addr.sa_len) : sizeof(struct ifreq))
                  ^
../netdev.c:339:10: note: in expansion of macro 'SIZEOF_ADDR_IFREQ'
   cnt += SIZEOF_ADDR_IFREQ(*ir);
          ^~~~~~~~~~~~~~~~~
make[7]: *** [netdev.o] Error 1
make[6]: *** [all-recursive] Error 1
Making all in doc
make[6]: Nothing to be done for `all'.
Making all in contrib
make[6]: Nothing to be done for `all'.
make[6]: Nothing to be done for `all-am'.
make[5]: *** [all-recursive] Error 1
make[4]: *** [all] Error 2
make[3]: *** [/Volumes/OpenWrt/openwrt/build_dir/target-aarch64_cortex-a72_musl/pdnsd-1.2.9b-par/.built] Error 2
make[3]: Leaving directory `/Volumes/OpenWrt/openwrt/package/lean/pdnsd-alt'
time: package/lean/pdnsd-alt/compile#4.72#3.52#9.84
make[2]: *** [package/lean/pdnsd-alt/compile] Error 2
make[2]: Leaving directory `/Volumes/OpenWrt/openwrt'
make[1]: *** [/Volumes/OpenWrt/openwrt/staging_dir/target-aarch64_cortex-a72_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory `/Volumes/OpenWrt/openwrt'
make: *** [world] Error 2

--- a/package/lean/pdnsd-alt/Makefile
+++ b/package/lean/pdnsd-alt/Makefile
@@ -38,7 +38,8 @@ TARGET_CFLAGS += -I$(STAGING_DIR)/usr/include
 CMAKE_OPTIONS += -DDEBUG=1
 
 CONFIGURE_ARGS += \
-		--with-cachedir=/var/pdnsd
+		--with-cachedir=/var/pdnsd \
+		--with-target=Linux
 
 define Package/pdnsd-alt/install
 	$(INSTALL_DIR) $(1)/usr/sbin

